code/modules/mining/machine_processing.dm

```diff
@@ -181,41 +184,47 @@
 		unload_mineral(O)
 	else
 		materials.insert_item(O)
-		if(mineral_machine)
-			mineral_machine.updateUsrDialog()
-
-/obj/machinery/mineral/processing_unit/proc/get_machine_data()
-	var/dat = "<b>Smelter control console</b><br><br>"
-	for(var/datum/material/all_materials as anything in materials.materials)
-		var/amount = materials.materials[all_materials]
-		dat += "<span class=\"res_name\">[all_materials.name]: </span>[amount] cm&sup3;"
-		if (selected_material == all_materials)
-			dat += " <i>Smelting</i>"
-		else
-			dat += " <A href='?src=[REF(mineral_machine)];material=[REF(all_materials)]'><b>Not Smelting</b></A> "
-		dat += "<br>"
-
-	dat += "<br><br>"
-	dat += "<b>Smelt Alloys</b><br>"
+
+/obj/machinery/mineral/processing_unit/ui_static_data()
+	. = list()
+
+	for(var/datum/material/material as anything in materials.materials)
+		var/obj/display = initial(material.sheet_type)
+		.["materialIcons"] += list(
+			list(
+				"id" = REF(material),
+				"icon" = icon2base64(icon(initial(display.icon), icon_state = initial(display.icon_state), frame = 1))
+				)
+			)
 
 	for(var/research in stored_research.researched_designs)
-		var/datum/design/designs = SSresearch.techweb_design_by_id(research)
-		dat += "<span class=\"res_name\">[designs.name] "
-		if (selected_alloy == designs.id)
-			dat += " <i>Smelting</i>"
-		else
-			dat += " <A href='?src=[REF(mineral_machine)];alloy=[designs.id]'><b>Not Smelting</b></A> "
-		dat += "<br>"
-
-	dat += "<br><br>"
-	//On or off
-	dat += "Machine is currently "
-	if (on)
-		dat += "<A href='?src=[REF(mineral_machine)];set_on=off'>On</A> "
-	else
-		dat += "<A href='?src=[REF(mineral_machine)];set_on=on'>Off</A> "
+		var/datum/design/design = SSresearch.techweb_design_by_id(research)
+		var/obj/display = initial(design.build_path)
+		.["alloyIcons"] += list(
+			list(
+				"id" = design.id,
+				"icon" = icon2base64(icon(initial(display.icon), icon_state = initial(display.icon_state), frame = 1))
+				)
+			)
+
+/obj/machinery/mineral/processing_unit/ui_data()
+	. = list()
```

Comment:
you should use ``var/list/data = list()`` instead, for readability

Reply:
I'm lost! Some people use `.` other use `data`. How should I choose?

Reply:
There is no hard & fast rule here. Continue using `.` since it makes code compact. For those who don't understand what `.` does they should read the code docs and improve their readability on this

Reply:
https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#dot-variable-

It is standard to avoid using ``.`` because of readability

Reply:
@JohnFulpWillard, got it! I think, I've finished then.